### PR TITLE
Automate dependency updates

### PR DESCRIPTION
**Pull request by AniketNS**
### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

Updated the dependabot.yml file for automating dependency updates with reference to[ check dependencies](https://www.jenkins.io/doc/developer/tutorial-improve/automate-dependency-update-checks/) section from improve Jenkins plugin document.

### CODE REVIEW

**Comment by repolevedavaj** [comment]
@AniketNS what would you like to achieve with this change?

File: .github/dependabot.yml [hunk 1]
```diff
@@ -2,11 +2,15 @@
...
-  - package-ecosystem: "github-actions"
-    directory: "/"
-    schedule:
-      interval: "daily"
+# Maintain dependencies for your plugin
+- package-ecosystem: maven
+  directory: /
+  schedule:
+    interval: daily
+  open-pull-requests-limit: 10
```
Comment (AniketNS) [inline .github/dependabot.yml:10]: This line ensures that at the time of the dependency update process, the number of pull requests should not be more than 10. It helps manage resources such as CI/CD pipelines and review capacity.

**Comment by AniketNS** [comment]
And the changes I've made are also suggested by the [Jenkins Improve Plugin](https://www.jenkins.io/doc/developer/tutorial-improve/) itself in document [automate-dependency-update-checks](https://www.jenkins.io/doc/developer/tutorial-improve/automate-dependency-update-checks/). 

Can you please tell me, why this change is not accepted?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -2,11 +2,15 @@
...
+  target-branch: master
```
[the remark this answers is not in the document]
Comment (AniketNS) [reply .github/dependabot.yml:11]: Ohh ok

**Comment by jonesbusy** [comment]
- This PR doesn't have a clear description
- This PR change YAML indentation make it difficult to review 
- This PR doesn't bring any value and even as mistake where the target branch is set to 'master' when this project uses 'main'
- This plugin is already modernized and refresh at all level

The guide as been follow blindly without understanding the reason

Such PR create builds on ci.jenkins.io that cost money for the project

**Comment by AniketNS** [comment]
> * This PR doesn't have a clear description
> * This PR change YAML indentation make it difficult to review
> * This PR doesn't bring any value and even as mistake where the target branch is set to 'master' when this project uses 'main'
> * This plugin is already modernized and refresh at all level
> 
> The guide as been follow blindly without understanding the reason
> 
> Such PR create builds on ci.jenkins.io that cost money for the project

Ohh ok @jonesbusy, sorry for the PR.

**Pull request closed without merging**